gui/mainwindow.py

```diff
@@ -1902,9 +1961,82 @@ def saveWindowGeometry(self):
             self.config_manager.save_window_geometry(size, position, False)
 
     def closeEvent(self, event):
+        if getattr(self, 'isDirty', False):
+            msg = QMessageBox(self)
+            msg.setWindowTitle("Unsaved Changes")
+            msg.setText("You have unsaved changes. Are you sure you want to exit without saving?")
+            pix = QPixmap(":/assets/openposter.png")
+            msg.setIconPixmap(pix.scaled(64, 64, Qt.KeepAspectRatio, Qt.SmoothTransformation))
+            msg.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
+            msg.setDefaultButton(QMessageBox.No)
+            reply = msg.exec()
+            if reply != QMessageBox.Yes:
+                event.ignore()
+                return
         self.saveSplitterSizes()
-        self.saveWindowGeometry() 
+        self.saveWindowGeometry()
         super().closeEvent(event)
 
     def openDiscord(self):
         webbrowser.open("https://discord.gg/t3abQJjHm6")
+    def saveFile(self):
+        if not (hasattr(self, 'cafilepath') and hasattr(self, 'cafile')):
+            return
+        save_path, _ = QFileDialog.getSaveFileName(
+            self,
+            "Save As",
+            self.cafilepath,
+            "Core Animation Bundle (*.ca)"
+        )
+        if not save_path:
+            return
+        dest_parent, base = os.path.split(save_path)
+        try:
+            self.cafile.write_file(base, dest_parent)
+            self.cafilepath = save_path
+            self.ui.filename.setText(save_path)
+            self.statusBar().showMessage(f"Saved As {save_path}", 3000)
+            self.isDirty = False
+            self.saveButton.setEnabled(False)
+        except Exception as e:
+            QMessageBox.critical(self, "Save Failed", f"Unable to save: {e}")
+
+    def markDirty(self):
+        self.isDirty = True
+        self.saveButton.setEnabled(True)
+
+    def onItemMoved(self, item):
+        layer_id = item.data(0)
+        layer = self.cafile.rootlayer.findlayer(layer_id)
+        if layer:
+            pos = item.pos()
+            layer.position = [str(pos.x()), str(pos.y())]
```

Comment:
honestly i forget if this is the correct format for that parameter or not (string or array). in the xml its like "100 100". when it is exported will it become "100 100" or "100, 100"